src/snowcli/cli/nativeapp/init.py

```diff
@@ -0,0 +1,291 @@
+    """
+    Could not complete rendering file from Jinja template.
+    """
+
+    # def __init__(self, message: str):
+    #     super.__init__(message)
+    def __init__(self, name: str):
+        super().__init__(
+            f"Could not complete rendering file from Jinja template: {name}"
+        )
+
+
+class CannotInitializeAnExistingProjectError(ClickException):
+    """
+    Cannot initialize a new project within an existing Native Application project.
+    """
+
+    def __init__(self):
+        super().__init__(self.__doc__)
+
+
+class DirectoryAlreadyExistsError(ClickException):
+    """
+    Directory already contains a project with the intended name
+    """
+
+    name: str
+
+    def __init__(self, name: str):
+        super().__init__(
+            f"This directory already contains a sub-directory called {name}. Please try a different name."
+        )
+        self.name = name
+
+
+def _sparse_checkout(
+    git_url: str, repo_sub_directory: str, target_parent_directory: str
+):
+    """
+    Clone the requested sub directory of a git repository from the provided git url.
+
+    Args:
+        git_url (str): The full git url to the repository to be cloned.
+        repo_sub_directory (str): The sub directory name within the repository to be cloned.
+        target_parent_directory (str): The parent directory where the git repository will be cloned into.
+
+    Returns:
+        None
+    """
+
+    clone_command = (
+        f"git clone -n --depth=1 --filter=tree:0 {git_url} {target_parent_directory}"
+    )
+    sparse_checkout_command = f"""
+        cd {target_parent_directory} &&
+            git sparse-checkout set --no-cone {repo_sub_directory} &&
+                git checkout
+        """
+    try:
+        subprocess.run(
+            clone_command,
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+        subprocess.run(
+            sparse_checkout_command,
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+    except subprocess.CalledProcessError as err:
+        log.error(err.stderr)
+        raise GitCloneError()
+
+
+def _move_and_rename_project(
+    source_parent_directory: Path,
+    target_parent_directory: Path,
+    repo_sub_directory: str,
+    new_name: str,
+):
+    """
+    Move the newly cloned repository's sub directory from its source directory to target directory, and rename to new_name.
+
+    Args:
+        source_parent_directory (Path): The source parent directory of the sub directory.
+        target_parent_directory (Path): The target parent directory for the sub directory.
+        repo_sub_directory (str): The sub directory name within the cloned repository.
+        new_name (str): The new name to give to the sub directory after moving.
+
+    Returns:
+        None
+    """
+
+    # Move to target parent directory
```

Comment:
1. Why this has to be done in two separate steps instead of moving the directory to target path (including target name) in just one step?
2. Why do we need custom function instead of `shutil.move`?